desc/protoparse/ast/source_pos.go

```diff
@@ -0,0 +1,30 @@
+package ast
+
+import "fmt"
+
+// SourcePos identifies a location in a proto source file.
+type SourcePos struct {
+	Filename  string
+	Line, Col int
+	Offset    int
+}
+
+func (pos SourcePos) String() string {
+	if pos.Line <= 0 || pos.Col <= 0 {
+		return pos.Filename
+	}
+	return fmt.Sprintf("%s:%d:%d", pos.Filename, pos.Line, pos.Col)
+}
+
+func UnknownPos(filename string) *SourcePos {
```

Comment:
Are these created in fallback, error scenarios?

Reply:
Yes. For example, if you import `google/protobuf/descriptor.proto` and don't provide your own copy, the protoparse package uses a version that is compiled in. But we don't have source for compiled in descriptors (in fact, not even source _info_ in the file descriptor, since those are stripped when `protoc-gen-go` embeds the descriptor bytes).